client/components/file-upload/index.tsx

```diff
@@ -41,6 +41,25 @@ export const FileUploadControl = ( {
 		/>
 	);
 
+	const handleButtonClick = (
+		event: React.MouseEvent< HTMLButtonElement >,
+		openFileDialog: () => void
+	) => {
+		// Get file input next to the button element and clear it's value, allowing to select the same file again in case of connection or general error or need to select it again
```

Comment:
I'd suggest leaving a comment here that links to https://github.com/WordPress/gutenberg/issues/39267, so that it's clear why this workaround is being used. That'll also make it clearer later when we might want to use `onChange` directly on `FormFileUpload` once it's supported.